lib/runner/cli/cli.js

```diff
@@ -138,16 +138,22 @@ class CliRunner {
   }
   
   setMobileOptions(argv) {
-    const {desiredCapabilities} = this.test_settings;
+    const {desiredCapabilities, selenium} = this.test_settings;
 
-    if (!isRealIos(desiredCapabilities)) {
-      return;
+    if (isRealIos(desiredCapabilities) && !selenium.use_appium) {
+      if (argv.deviceId) {
+        this.test_settings.desiredCapabilities['safari:deviceUDID'] = iosRealDeviceUDID(argv.deviceId);
+      } else if (!desiredCapabilities['safari:deviceUDID']) {
+        throw new RealIosDeviceIdError();
+      }
     }
 
-    if (argv.deviceId) {
-      this.test_settings.desiredCapabilities['safari:deviceUDID'] = iosRealDeviceUDID(argv.deviceId);
-    } else if (!desiredCapabilities['safari:deviceUDID']) {
-      throw new RealIosDeviceIdError();
+    if (isAndroid(desiredCapabilities) && argv.deviceId) {
```

Comment:
We should check for Appium server here as well, since we do not want to set `androidDeviceSerial` in case we are using Appium. We set `appium:udid` capability in that case.

Reply:
@harshit-bs any thoughts regarding this?

Reply:
Check added in [83a9e00](https://github.com/nightwatchjs/nightwatch/pull/3572/commits/83a9e00c0c680dd07b9ac8fac6f55835a9bb1acd)